src/compute/src/render/reduce.rs

```diff
@@ -1058,54 +1020,54 @@ where
 
     // Instantiate a default vector for diffs with the correct types at each
     // position.
-    let zero_diffs: Vec<_> = full_aggrs
-        .iter()
-        .map(|f| {
-            let inner = match f.func {
-                AggregateFunc::Any | AggregateFunc::All => AccumInner::Bool {
+    let zero_diffs: (Vec<_>, Diff) = (
+        full_aggrs
+            .iter()
+            .map(|f| match f.func {
+                AggregateFunc::Any | AggregateFunc::All => Accum::Bool {
                     trues: 0,
                     falses: 0,
                 },
-                AggregateFunc::SumFloat32 | AggregateFunc::SumFloat64 => AccumInner::Float {
+                AggregateFunc::SumFloat32 | AggregateFunc::SumFloat64 => Accum::Float {
                     accum: 0,
                     pos_infs: 0,
                     neg_infs: 0,
                     nans: 0,
                     non_nulls: 0,
                 },
-                AggregateFunc::SumNumeric => AccumInner::Numeric {
+                AggregateFunc::SumNumeric => Accum::Numeric {
                     accum: OrderedDecimal(NumericAgg::zero()),
                     pos_infs: 0,
                     neg_infs: 0,
                     nans: 0,
                     non_nulls: 0,
                 },
-                _ => AccumInner::SimpleNumber {
+                _ => Accum::SimpleNumber {
                     accum: 0,
                     non_nulls: 0,
                 },
-            };
-            Accum { inner, total: 0 }
-        })
-        .collect();
+            })
+            .collect(),
+        0,
+    );
 
     // Two aggregation-specific values for each aggregation.
     let datum_to_accumulator = move |datum: Datum, aggr: &AggregateFunc| {
-        let inner = match aggr {
-            AggregateFunc::Count => AccumInner::SimpleNumber {
+        match aggr {
+            AggregateFunc::Count => Accum::SimpleNumber {
                 accum: 0, // unused for AggregateFunc::Count
                 non_nulls: if datum.is_null() { 0 } else { 1 },
             },
             AggregateFunc::Any | AggregateFunc::All => match datum {
-                Datum::True => AccumInner::Bool {
+                Datum::True => Accum::Bool {
                     trues: 1,
                     falses: 0,
                 },
-                Datum::Null => AccumInner::Bool {
+                Datum::Null => Accum::Bool {
```

Comment:
The removed comment above states

> It is important that `Accum.is_zero()` returns `false` if at least one null record has been observed. 

but wouldn't this no longer happen in this case?

Reply:
Yes it is no longer important, afaict. Elsewhere, `total` will be greater than zero when a non-null record has been observed, and that is what feeds into the production of `Null` for the transforms that previously relied on this.

Reply:
Ok, LGTM then if that's the case